front/components/poke/PokeSlackChannelPatternInput.tsx

```diff
@@ -0,0 +1,101 @@
+import { ioTsResolver } from "@hookform/resolvers/io-ts";
+import * as t from "io-ts";
+import React from "react";
+import { FormProvider, useForm } from "react-hook-form";
+
+import { PokeButton } from "@app/components/poke/shadcn/ui/button";
+import { InputField } from "@app/components/poke/shadcn/ui/form/fields";
+import { useSubmitFunction } from "@app/lib/client/utils";
+
+interface SlackChannelPatternInputProps {
+  initialValue: string | null;
+  ownerSId: string;
+}
+
+export const SlackChannelFormSchema = t.type({
+  inputValue: t.union([t.string, t.null]),
+});
+
+export type SlackChannelFormType = t.TypeOf<typeof SlackChannelFormSchema>;
+
+export function SlackChannelPatternInput({
+  initialValue,
+  ownerSId,
+}: SlackChannelPatternInputProps) {
+  const formMethods = useForm<SlackChannelFormType>({
+    resolver: ioTsResolver(SlackChannelFormSchema),
+    defaultValues: {
+      inputValue: initialValue,
+    },
+  });
+
+  const { submit: handleWhiteListedChannelPatternChange } = useSubmitFunction(
+    async (newValue: string | null) => {
+      try {
+        const r = await fetch(
+          `/api/poke/workspaces/${ownerSId}/data_sources/managed-slack/config`,
+          {
+            method: "POST",
+            headers: {
+              "Content-Type": "application/json",
+            },
+            body: JSON.stringify({
+              configKey: "whiteListedChannelPattern",
+              configValue: newValue,
```

Comment:
Do we need to support `null` here to remove the pattern? If so you need to update here as well I presume:

https://github.com/dust-tt/dust/blob/3687267c63af97c1777bc463440685758eef7c78/connectors/src/api/connector_config.ts#L14-L16

Reply:
Indeed no need to support `null` here

Reply:
We do want to support null no?